### DISCUSSION TITLE
Simple Run-Once Setup

### DISCUSSION THREAD

**Discussion by alec-hs** [coder/code-server#3579] [Show and tell]
I really like using Coder when on the move and only have an iPad or tablet with me but don't always want to keep a VM running so I made a script that will get Coder up and running in a few mins with minimal input. 

[Coder Cloudflare Setup](https://github.com/alec-hs/coder-cloudflare-setup)

I have used and tested this with Hetzner Cloud VMs but it should work anywhere, it features the following:

- Single command setup
- HTTPS using Cloudflare and Lets Encrypt
- Set GUI Password
- Caddy Reverse Proxy

Current tested and supported Distros are:

- Debian 10
- Ubuntu 16.04.7 LTS

Any feedback, ideas or suggestions welcome.

**Comment by jsjoeio** [comment]
This is awesome! Thanks for sharing 😄

cc @bpmct

**Comment by bpmct** [comment]
This is awesome! I see that is solves a dns01 challenge, this is what is needed to also generate LetsEncrypt certificates. This would be perfect for setting a subdomain to proxy ports: https://github.com/cdr/code-server/blob/main/docs/FAQ.md#sub-domains.

While ports can be proxied via /proxy/[port], it is less reliable as many services like React are picky and want a root path. Would you be able to add the subdomain aspect to this? I created an issue: https://github.com/alec-hs/coder-cloudflare-setup/issues/1

While the wildcard traffic can't be proxied under cloudflare (grey cloud) it can still be secure with the server's LE certificate.

**Reply by bpmct** [reply]
I'll also mention this project in [deploy-code-server](https://github.com/cdr/deploy-code-server) under the "VMs" section

**Reply by alec-hs** [reply]
All done and added. Thanks for the help with the sub domain issue.

**Reply by bpmct** [reply]
@alec-hs - we are working on [v2.coder.com](https://v2.coder.com) and was wondering if you'd like early access. We'd appreciate your feedback

**Reply by alec-hs** [reply]
Sure thing, that'd be great. Just signed up using the form on the site.

**Reply by bpmct** [reply]
Great I just sent you an invite. Feel free to join [our Discord](https://discord.gg/coder) too and we can chat there
